integration/presto/src/main/prestosql/org/apache/carbondata/presto/CarbondataPageSourceProvider.java

```diff
@@ -58,37 +60,40 @@
   private HdfsEnvironment hdfsEnvironment;
 
   @Inject public CarbondataPageSourceProvider(
+      TypeManager typeManager,
       HiveConfig hiveConfig,
       HdfsEnvironment hdfsEnvironment,
-      Set<HiveRecordCursorProvider> cursorProviders,
       Set<HivePageSourceFactory> pageSourceFactories,
-      TypeManager typeManager,
+      Set<HiveRecordCursorProvider> cursorProviders,
+      GenericHiveRecordCursorProvider genericCursorProvider,
       CarbonTableReader carbonTableReader) {
-    super(hiveConfig, hdfsEnvironment, cursorProviders, pageSourceFactories, typeManager);
+    super(typeManager, hiveConfig, hdfsEnvironment, pageSourceFactories, cursorProviders,
+        genericCursorProvider);
     this.carbonTableReader = requireNonNull(carbonTableReader, "carbonTableReader is null");
     this.hdfsEnvironment = hdfsEnvironment;
   }
 
   @Override
-  public ConnectorPageSource createPageSource(ConnectorTransactionHandle transactionHandle,
+  public ConnectorPageSource createPageSource(ConnectorTransactionHandle transaction,
       ConnectorSession session, ConnectorSplit split, ConnectorTableHandle table,
-      List<ColumnHandle> columns) {
-    HiveSplit carbonSplit =
-        checkType(split, HiveSplit.class, "split is not class HiveSplit");
+      List<ColumnHandle> columns, TupleDomain<ColumnHandle> dynamicFilter) {
+    HiveSplit carbonSplit = checkType(split, HiveSplit.class, "split is not class HiveSplit");
     this.queryId = carbonSplit.getSchema().getProperty("queryId");
     if (this.queryId == null) {
       // Fall back to hive pagesource.
-      return super.createPageSource(transactionHandle, session, split, table, columns);
+      return super.createPageSource(transaction, session, split, table, columns, dynamicFilter);
     }
+    // TODO: check and use dynamicFilter in CarbondataPageSource
```

Comment:
if possible, can you create jira for this and add the jira in TODO?

Reply:
https://issues.apache.org/jira/browse/CARBONDATA-4103